conans/test/integration/generators/cpp_info_set_generator_properties_test.py

```diff
@@ -289,6 +289,95 @@ def package_info(self):
     assert find_package_contents_old == find_package_contents
 
 
+@pytest.mark.parametrize("generator", ["cmake_find_package_multi", "cmake_find_package"])
+def test_cmake_find_package_target_namespace(generator):
+    # https://github.com/conan-io/conan/issues/9946
+    client = TestClient()
+    hello = textwrap.dedent("""
+        import os
+        from conans import ConanFile
+        class MyHello(ConanFile):
+            settings = "build_type"
+            name = "hello"
+            version = "1.0"
+            def package_info(self):
+                self.cpp_info.components["helloworld"].set_property("cmake_target_name", "HelloWorld")
+                {}
+
+        """)
+
+    greetings = textwrap.dedent("""
+        import os
+        from conans import ConanFile
+        class MyPkg(ConanFile):
+            settings = "build_type"
+            name = "greetings"
+            version = "1.0"
+            requires = "hello/1.0"
+            def package_info(self):
+                self.cpp_info.components["greetingshello"].requires = ["hello::helloworld"]
+                {}
+        """)
+
+    client.save({"hello.py": hello.format('self.cpp_info.set_property("cmake_target_namespace", "hello_namespace")'),
```

Comment:
can we also check `cmake_target_namespace` property works on component level?

Reply:
From the documentation https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html#properties I think it should not work at component level, just at the root cpp_info level.

Reply:
I think that if you want to change the namespace at component level you should use the alias feature right?

Reply:
for sure, you may use an alias. it's just misleading and confusing for me that some properties work on component level, but some don't.